test/unit/version_test.py

```diff
@@ -0,0 +1,207 @@
+            return self._get_subprocess_run_mock(args)
+
+        with patch("subprocess.run", side_effect=simulate_fail):
+            with pytest.raises(ReleaseError) as ex:
+                _trigger_release()
+        assert str(error_cmd) in str(ex)
+
+    def test_default_branch_could_not_be_found(self, mock_from_poetry):
+        def simulate_fail(args, **kwargs):
+            if args == ("git", "remote", "show", "origin"):
+                return MagicMock(returncode=0, stdout="DUMMY TEXT")
+            return self._get_subprocess_run_mock(args)
+
+        with patch("subprocess.run", side_effect=simulate_fail):
+            with pytest.raises(ReleaseError) as ex:
+                _trigger_release()
+        assert "default branch could not be found" in str(ex)
+
+    def test_tag_already_exists(self, mock_from_poetry):
+        version = mock_from_poetry.return_value
+
+        def simulate_fail(args, **kwargs):
+            if args == ("git", "tag", "--list"):
+                return MagicMock(returncode=0, stdout=f"0.1.0\n0.2.0\n{version}")
+            return self._get_subprocess_run_mock(args)
+
+        with patch("subprocess.run", side_effect=simulate_fail):
+            with pytest.raises(ReleaseError) as ex:
+                _trigger_release()
+        assert f"tag {version} already exists" in str(ex)
+
+    def test_release_already_exists(self, mock_from_poetry):
+        version = mock_from_poetry.return_value
+
+        def simulate_fail(args, **kwargs):
+            if args == ("gh", "release", "list"):
+                return MagicMock(returncode=0, stdout=f"0.1.0\n0.2.0\n{version}")
+            return self._get_subprocess_run_mock(args)
+
+        with patch("subprocess.run", side_effect=simulate_fail):
+            with pytest.raises(ReleaseError) as ex:
+                _trigger_release()
+        assert f"release {version} already exists" in str(ex)
+
+
+@patch("exasol.toolbox.util.version.which", return_value=None)
+def test_poetry_decorator_no_poetry_executable(mock):
+    @poetry_command
+    def test():
+        pass
+
+    with pytest.raises(ToolboxError):
+        test()
+
+
+@patch("exasol.toolbox.util.version.which", return_value="test/path")
+def test_poetry_decorator_subprocess(mock):
+    @poetry_command
+    def test():
+        raise subprocess.CalledProcessError(returncode=1, cmd=["test"])
+        pass
+
+    with pytest.raises(ToolboxError):
+        test()
+
+
+def test_version_from_python_module(tmp_path):
+    tmp_file = tmp_path / "file"
+    file = """
+MAJOR = 1
+MINOR = 2
+PATCH = 3
+VERSION = f"{MAJOR}.{MINOR}.{PATCH}"
+__version__ = VERSION
+    """
+    tmp_file.write_text(file)
+    assert Version.from_python_module(tmp_file) == Version.from_string("1.2.3")
+
+
+def test_version_from_python_no_module_error(tmp_path):
+    file_path = tmp_path / "file"
+    file_path.write_text("")
+    with pytest.raises(ToolboxError) as ex:
```

Comment:
currently, you aren't testing the value of `ex`, so it can be removed. you could check to see if (part of) a string matched the text, like `assert "version within module" in str(ex.value)`.

There's a debate about checking error message strings. Some folks say it's useful especially when ToolboxError could be tossed at a multitude of spots to verify that it you got the error at the place you expected it & not someone else. Whereas, others think it can be problematic, as when you update the string text, you'll need to update the text. I'm a bit more in the first group, but I'll let you decide what you prefer to do here.

```suggestion
    with pytest.raises(ToolboxError):
```